Public/Src/Cache/MemoizationStore/InterfacesTest/Sessions/ContentHashListTests.cs

```diff
@@ -136,5 +137,47 @@ public void RandomNullPayload()
         {
             Assert.Null(ContentHashList.Random().Payload);
         }
+
+        [Fact]
+        public void SerializeRoundtrip()
+        {
+            var v = ContentHashList.Random();

```

Comment:
I expect to see similar tests for a bunch of types, so I would say that we should have a helper to test serialization roundtrip.

A helper that will take two callbacks to serailize and deserialize the value and the helper will do all the stream management and assertions in it.